### PR TITLE
feat: cherry-pick squash commit

### PR DESCRIPTION
**Pull request by codebytere**
Closes https://github.com/electron/trop/issues/185.

Cherry pick only the merge commit instead of each individual commit.

cc @MarshallOfSound

### CODE REVIEW

**Review by jkleinsc — APPROVED** [review]

**Review by ckerr — APPROVED** [review]
I love the idea of having this as an option & IMO for it being the default.

@MarshallOfSound had some discussion in the bug ticket about not removing the previous behavior but instead having the two features side-by-side, so that q should be resolved before merging despite the :heavy_check_mark:

**Comment by codebytere** [comment]
@MarshallOfSound reworked this a bit

**Review by ckerr — APPROVED** [review]
re-approving with the new changes

File: src/utils.ts [hunk 1]
```diff
@@ -65,6 +66,148 @@ export const labelClosedPR = async (
...
+      patches[i] = await patchBody.text();
+      log(
+        'backportImpl',
+        LogLevel.INFO,
+        `Got patch (${i + 1}/${commits.length})`,
+      );
+    });
+  }
+
+  await new Promise((r) => q.start(r));
```
Comment (MarshallOfSound) [inline src/utils.ts:138]: I know this is existing code and I probably wrote it but it's wrong 😆 

```suggestion
  await new Promise((resolve, reject) => q.start(err ? reject(err) : resolve()));
```

File: src/utils.ts [hunk 2]
```diff
@@ -65,6 +66,148 @@ export const labelClosedPR = async (
...
+  }
+
+  return success;
+};
+
+const tryBackportSquashCommit = async (opts: TryBackportOptions) => {
+  // Fetch the merged squash commit.
+  log('backportImpl', LogLevel.INFO, `Fetching squash commit details`);
+
+  const patchUrl = `https://api.github.com/repos/${opts.slug}/commits/${opts.pr.merge_commit_sha}`;
```
Comment (MarshallOfSound) [inline src/utils.ts:172]: We should only do this if `opts.pr.merged`, otherwise `merge_commit_sha` isn't a squashed commit

**Review by MarshallOfSound — APPROVED** [review]

**Pull request merged**